### PR TITLE
ONI Cargo Fixes and Reduction of Equipment in Vendors

### PR DESCRIPTION
**Pull request by Aroliacue**
:cl: Aroliacue
rscadd: Added new ONI category Restricted Goods with special purchaseables for research.
tweak: Fixed ONI Cargo not showing up.
tweak: Lowered counts of items in vendors to 30 from 50.
tweak: Adjusted value of tradeable items.
tweak: Reduced cargo income amounts. Funds can still be made via NPC traders.
/:cl:

### CODE REVIEW

File: code/modules/halo/weapons/covenant/snipers.dm [hunk 1]
```diff
@@ -20,6 +20,7 @@
 	accuracy = 1
 	wielded_item_state = "carbine-wielded"
 	speed_reload_time = -1 //already auto-ejects
+	salvage_components = list(/obj/item/plasma_core)
```
Comment (X0-11) [inline code/modules/halo/weapons/covenant/snipers.dm]: the cov carbine doesnt use plasma

File: code/modules/halo/weapons/covenant/snipers.dm [hunk 2]
```diff
@@ -127,6 +128,7 @@
 	accuracy = -1
 	scoped_accuracy = 2
 	wielded_item_state = "needlerifle-wielded"
+	salvage_components = list(/obj/item/plasma_core)
```
Comment (X0-11) [inline code/modules/halo/weapons/covenant/snipers.dm]: needlerifle doesnt use plasma

File: code/modules/halo/weapons/covenant/pistols.dm [hunk 3]
```diff
@@ -148,6 +148,7 @@
 	hud_bullet_reffile = 'code/modules/halo/icons/hud_display/hud_bullet_5x6.dmi'
 	hud_bullet_iconstate = "needle"
 	slowdown_general = 0
+	salvage_components = list(/obj/item/plasma_core)
```
Comment (X0-11) [inline code/modules/halo/weapons/covenant/pistols.dm]: needler doesnt use plasma

File: code/modules/halo/weapons/covenant/launchers.dm [hunk 4]
```diff
@@ -27,7 +27,7 @@
 		)
 	crosshair_file = 'code/modules/halo/weapons/icons/dragaim_missile.dmi'
 	wielded_item_state = "fuelrod-wielded"
-	salvage_components = list(/obj/item/plasma_core)
+	salvage_components = list(/obj/item/plasma_core = 3)
```
Comment (X0-11) [inline code/modules/halo/weapons/covenant/launchers.dm]: frc doesnt use plasma

File: code/modules/halo/weapons/covenant/brute.dm [hunk 5]
```diff
@@ -31,6 +31,7 @@
 	num_fragments = 250 //50 more than a high yield frag bomb
 
 	lunge_dist = 3
+	salvage_components = list(/obj/item/plasma_core)
```
Comment (X0-11) [inline code/modules/halo/weapons/covenant/brute.dm:34]: most brute weapons dont use plasma

File: code/modules/halo/weapons/covenant/brute.dm [hunk 6]
```diff
@@ -82,6 +83,7 @@
 	"Jiralhanae" = 'code/modules/halo/covenant/species/jiralhanae/jiralhanae_weapons_big.dmi'
 	)
 	lunge_dist = 3
+	salvage_components = list(/obj/item/plasma_core)
```
Comment (X0-11) [inline code/modules/halo/weapons/covenant/brute.dm]: grenades probably shouldn't have plasma cores

**Review by X0-11 — CHANGES_REQUESTED** [review]

**Comment by Aroliacue** [comment]
What do they gain from deconstructing these weapons then?

**Comment by X0-11** [comment]
Something not currently implemented. Leave space open for further expansion.

**Comment by Aroliacue** [comment]
No worries, updating soon

**Comment by Aroliacue** [comment]
Reduced cargo income amounts and removed plasma cores from non-plasma weapons

**Review by X0-11 — APPROVED** [review]

File: code/modules/halo/factions/player_factions.dm [hunk 7]
```diff
@@ -23,7 +23,7 @@
 		/mob/living/simple_animal/hostile/covenant/drone/ranged = 5)
 	default_radio_channel = RADIO_COV
 	special_jobs = list(/datum/job/covenant/mgalekgolo)
-	income = 1000
+	income = 600 //Higher than UNSC/ONI as they don't have an easy way to make money and also don't have purchases split
```
Comment (X0-11) [inline code/modules/halo/factions/player_factions.dm:26]: i dont think this quite makes up for the double income of 2x400 +supplementary from trading tbh. drop unsc / oni to like 100 and this makes a bit more sense. unsc/oni split doesnt mean much if they just extract the credits and deposit them in the other account

File: code/modules/halo/factions/player_factions.dm [hunk 8]
```diff
@@ -96,7 +96,7 @@
 		/mob/living/simple_animal/hostile/battledog/odst = 1,\
 		/mob/living/simple_animal/hostile/unsc/spartan_two = 1)
 	default_radio_channel = RADIO_ONI
-	income = 1000
+	income = 400
```
Comment (X0-11) [inline code/modules/halo/factions/player_factions.dm:99]: The URF probably deserve some sort of income, however minor.

File: code/modules/halo/factions/player_factions.dm (hunk 7 above)
Comment (X0-11) [reply code/modules/halo/factions/player_factions.dm:26]: We could also make them more similar and give the covenant a set of traders for salvaged human equipment (you could make neural laces give a fair chunk of money too)

Comment (Aroliacue) [reply code/modules/halo/factions/player_factions.dm:26]: Covvies do have Kig-Yar traders in code, albeit I believe they only accept Covenant equipment and mined/farmed goods. There's a few legacy rooms on the CRS, such as the bow chapel/storage areas that I want to convert over into a new cargo setup, and it'd be easy to maybe stuff in a small botany bay and trader somewhere there (flavoured as specimens for agricultural support ships in the fleet?) I was asleep and missed this, I'll update the incomes on the second passover.

**Pull request merged**